applications/tari_console_wallet/src/automation/commands.rs

```diff
@@ -253,6 +275,17 @@ pub async fn coin_split(
     Ok(tx_id)
 }
 
+pub fn sign_message(private_key: String, challenge: String) -> Result<(Signature, RistrettoSecretKey), CommandError> {
+    let private_key =
+        PrivateKey::from_hex(private_key.as_str()).map_err(|e| CommandError::InvalidArgument(e.to_string()))?;
+    let challenge = challenge.as_bytes();
+
+    let nonce = PrivateKey::random(&mut OsRng);
+    let signature = Signature::sign(private_key, nonce.clone(), challenge).map_err(CommandError::FailedSignature)?;
+
+    Ok((signature, nonce))
```

Comment:
We need to output the public nonce here, not the private one. 
Signatures are:
`sig = r + e*k`
and you publish: `sig and R`, which is the public version of `r`

Reply:
Thanks for the explanation ! I have refactored the code accordingly .